### PR TITLE
make netbricks the default vagrant

### PR DESCRIPTION
**Pull request by drunkirishcoder**
It's a little bit awkward to ssh into one that's not netbricks. `DIMAGE=sandbox vagrant ssh`.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -52,7 +59,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # VirtualBox-specific configuration
   config.vm.provider "virtualbox" do |vb|
     # Set machine name, memory and CPU limits
-    vb.name = "ubuntu-xenial-williamofockham"
+    vb.name = "ubuntu-xenial-#{$dimage}"
```
Comment (seancribbs) [inline Vagrantfile]: We shouldn't change the VM name based on an environment variable.

File: Vagrantfile [hunk 2]
```diff
@@ -21,14 +28,14 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
   # Setup the VM for DPDK, including binding the extra interface via the fetched
   # container
-  config.vm.provision "shell", path: "vm-setup.sh", args: $dimage
+  config.vm.provision "shell", path: "vm-setup.sh", args: $dtag
```
Comment (seancribbs) [inline Vagrantfile]: I don't know why this has arguments at all, it doesn't use them underneath.

File: Vagrantfile [hunk 3]
```diff
@@ -12,7 +16,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.box = "ubuntu/xenial64"
   config.disksize.size = "30GB"
   config.vm.synced_folder ".", "/vagrant", disabled: false
-  config.vm.synced_folder "../MoonGen", "/MoonGen", disabled: false
+  config.vm.synced_folder $nbpath, "/NetBricks", disabled: false
+  config.vm.synced_folder $mgpath, "/MoonGen", disabled: false
+
+  config.vm.define $dimage
```
Comment (seancribbs) [inline Vagrantfile]: Again, this shouldn't be dependent on an environment variable, it'll make Vagrant really messed up.

Comment (drunkirishcoder) [reply Vagrantfile]: Should I just define 4 different vagrant boxes, one for each docker image?

Comment (seancribbs) [reply Vagrantfile]: No, we should be able to get by with a single vagrant for now. Let's only create them when absolutely needed.

**Comment by drunkirishcoder** [comment]
Check for path before syncing them. Also to use a different docker image, just reprovision the vagrant with `DIMAGE=sandbox vagrant provision`.

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder @seancribbs approved this as is, but as we discussed on rabble today, do we want to not fix the device/port ids that we bind to, and instead default to those two and allow for a variable pass-in?

**Comment by zeeshanlakhani** [comment]
👍

**Pull request merged**